numpy/core/src/umath/ufunc_object.c

```diff
@@ -762,16 +767,17 @@ get_ufunc_arguments(PyUFuncObject *ufunc,
         if (!strcmp(ufunc_name, "equal") ||
                 !strcmp(ufunc_name, "not_equal")) {
             /* Warn on non-scalar, return NotImplemented regardless */
-            assert(nin == 2);
```

Comment:
Any reason for this change?

Reply:
I moved `nin == 2` inside the `if` statement above, so that ufuncs with other numbers of inputs are not checked. With the check just above, the assert seemed superfluous.